reV/SAM/generation.py

```diff
@@ -624,6 +625,256 @@ def default(self):
+        df['DNI'] = df.dni
+        df['DHI'] = df.dhi
+        df['Dew Point'] = df.dew_point
+        df['Temperature'] = df.air_temperature
+        df['Pressure'] = df.surface_pressure
+        df['Wind Speed'] = df.wind_speed
+        df = df.drop(['dt', 'dni', 'dhi', 'wind_speed', 'air_temperature',
+                      'dew_point', 'surface_pressure'], axis=1)
+        df.to_csv(fname, index=False, mode='a')
+        assert df.shape[0] == 8760 * 2
+
+        return fname
+
+    def _gen_exec(self, delete_wfile=True):
+        """
+        Run SAM generation with possibility for follow on econ analysis.
+
+        Parameters
+        ----------
+        delete_wfile : bool
+            Delete PySAM weather file after processing is complete
+        """
+        super()._gen_exec()
+
+        if delete_wfile:
+            logger.debug('Removing PySAM weather file')
+            os.remove(self._pysam_w_fname)
+
+
+class SolarWaterHeat(SolarThermal):
+    """
+    Solar Water Heating generation
+    """
+    MODULE = 'solarwaterheat'
+    PYSAM = pysam_swh
+
+    def __init__(self, resource=None, meta=None, parameters=None,
+                 output_request=None):
+        """Initialize a SAM solar water heating object.
+        """
+        self._pysam_weather_tag = 'solar_resource_file'
+        super().__init__(resource=resource, meta=meta, parameters=parameters,
+                         output_request=output_request)
+
+    @property
+    def default(self):
+        """Get the executed default pysam swh object.
+        Returns
+        -------
+        _default : PySAM.
+            Executed  pysam object.
+        """
+        if self._default is None:
+            res_file = os.path.join(
+                DEFAULTSDIR,
+                'SAM/USA AZ Phoenix Sky Harbor Intl Ap (TMY3).csv')
+            self._default = pysam_swh.default('SolarWaterHeatingNone')
+            self._default.Weather.solar_resource_file = res_file
+            self._default.execute()
+        return self._default
+
+
+class LinearDirectSteam(SolarThermal):
+    """
+    Process heat linear direct steam generation
+    """
+    MODULE = 'lineardirectsteam'
+    PYSAM = pysam_lfdi
+
+    def __init__(self, resource=None, meta=None, parameters=None,
+                 output_request=None):
+        """Initialize a SAM process heat liner direct steam object.
+        """
+        self._pysam_weather_tag = 'file_name'
+        super().__init__(resource=resource, meta=meta, parameters=parameters,
+                         output_request=output_request)
+
+    def cf_mean(self):
+        """Calculate mean capacity factor (fractional) from SAM.
+
+        Returns
+        -------
+        output : float
+            Mean capacity factor (fractional).
+        """
+        net_power = self['annual_field_energy'] \
+            - self['annual_thermal_consumption']  # kW-hr
+        name_plate = self['q_pb_des'] * 8760 * 1000  # q_pb_des is in MW
```

Comment:
q_pb_des is a scalar value? Not an array? *8760 is just to go from MW to MWh right?